owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/capabilities/OCCapabilityViewModel.kt

```diff
@@ -0,0 +1,106 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David González Verdugo
+ * @author Abel García de Prada
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.viewmodels.capabilities
+
+import android.accounts.Account
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.owncloud.android.data.capabilities.db.OCCapabilityEntity
+import com.owncloud.android.domain.capabilities.usecases.CapabilitiesLiveDataUseCase
+import com.owncloud.android.domain.capabilities.usecases.RefreshCapabilitiesUseCase
+import com.owncloud.android.operations.common.OperationType
+import com.owncloud.android.presentation.UIResult
+import com.owncloud.android.ui.errorhandling.ErrorMessageAdapter
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * View Model to keep a reference to the capability repository and an up-to-date capability
+ */
+class OCCapabilityViewModel(
+    val context: Context,
+    val account: Account,
+    capabilitiesLiveDataUseCase: CapabilitiesLiveDataUseCase = CapabilitiesLiveDataUseCase(context, account),
+    private val refreshCapabilitiesUseCase: RefreshCapabilitiesUseCase = RefreshCapabilitiesUseCase(context, account)
+) : ViewModel() {
+
+    private val _capabilities = MutableLiveData<UIResult<OCCapabilityEntity>>()
+    val capabilities: LiveData<UIResult<OCCapabilityEntity>> = _capabilities
+
+    private var capabilitiesLiveData: LiveData<OCCapabilityEntity>? = capabilitiesLiveDataUseCase.execute(
+        CapabilitiesLiveDataUseCase.Params(
+            accountName = account.name
+        )
+    ).data
+
+    // to detect changes in capabilities
+    private val capabilitiesObserver: Observer<OCCapabilityEntity> = Observer { capabilities ->
+        if (capabilities != null) {
+            _capabilities.postValue(UIResult.success(capabilities))
+        }
+    }
+
+    init {
+        capabilitiesLiveData?.observeForever(capabilitiesObserver)
+
+        refreshCapabilitiesFromNetwork()
+    }
+
+    private fun refreshCapabilitiesFromNetwork() {
+        viewModelScope.launch {
```

Comment:
inject the dispatcher where you want to run it in

I recommend you to read this post

https://medium.com/androiddevelopers/testing-two-consecutive-livedata-emissions-in-coroutines-5680b693cbf8